main/lsp/LSPTypechecker.cc

```diff
@@ -217,12 +217,23 @@ vector<core::FileRef> LSPTypechecker::runFastPath(LSPFileUpdates &updates, Worke
     Timer timeit(config->logger, "fast_path");
     // Replace error queue with one that is owned by this thread.
     gs->errorQueue = make_shared<core::ErrorQueue>(gs->errorQueue->logger, gs->errorQueue->tracer, errorFlusher);
-    auto result = updates.fastPathFilesToTypecheck(*gs, *config);
-    config->logger->debug("Added {} files that were not part of the edit to the update set", result.extraFiles.size());
+
+    std::vector<core::FileRef> toTypecheck;
+    toTypecheck.reserve(updates.fastPathExtraFiles.size() + updates.fastPathChangedFiles.size());
+    for (auto &path : updates.fastPathExtraFiles) {
+        auto fref = gs->findFileByPath(path);
+        ENFORCE(fref.exists());
+        toTypecheck.emplace_back(fref);
+    }
+
+    config->logger->debug("Added {} files that were not part of the edit to the update set", toTypecheck.size());
     UnorderedMap<core::FileRef, core::FoundDefHashes> oldFoundHashesForFiles;
-    auto toTypecheck = move(result.extraFiles);
-    auto shouldRunIncrementalNamer = !result.changedSymbolNameHashes.empty();
-    for (auto [fref, idx] : result.changedFiles) {
+    auto shouldRunIncrementalNamer = updates.fastPathUseIncrementalNamer;
+    UnorderedSet<core::FileRef> changedFiles;
+    for (auto [path, idx] : updates.fastPathChangedFiles) {
```

Comment:
Should this be a reference?
```suggestion
    for (auto &[path, idx] : updates.fastPathChangedFiles) {
```